### PR TITLE
make smart graphs transaction-aware

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Make smart graphs transaction-aware too.
Enterprise companion PR: https://github.com/arangodb/enterprise/pull/308

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6053/

### CODE REVIEW

File: arangod/Transaction/Manager.cpp [hunk 1]
```diff
@@ -391,6 +395,29 @@ Result Manager::createManagedTrx(TRI_vocbase_t& vocbase, TRI_voc_tid_t tid,
                   std::string(TRI_errno_string(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND)) +
                       ":" + cname);
       } else {
+#ifdef USE_ENTERPRISE
```
Comment (graetzer) [inline arangod/Transaction/Manager.cpp]: seriously ?

Comment (jsteemann) [reply arangod/Transaction/Manager.cpp]: Tja, is there a better alternative for this?

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/6089/

**Comment by Simran-B** [comment]
I will update the known issue as follows:
> Stream Transactions do not support the graph operations that are initiated via the `general-graph` / `smart-graph` JavaScript module or via the REST API at `/_api/gharial`. These operations will act as if no stream transaction is present.
>
> Fixed in versions: 3.5.1

**Comment by jsteemann** [comment]
Thanks @simran-b!

**Pull request merged**